fails to compile cpp file bundled in windows env

windows 環境で oj-bundle を走らせた際に、生成コード上の #line ディレクティブ でパスの区切り文字が '\\' (バックスラッシュ)になり、コンパイルに通らなくなってしまう。

error log

``` console
C:\Users\kimiyuki\Desktop\New folder>oj-bundle foo.cpp
INFO:onlinejudge_verify.config:no config file
DEBUG:onlinejudge_verify.languages.cplusplus_bundle:foo.cpp: line 1: #include "../foo.h"
#line 1 "C:\Users\kimiyuki\Desktop\foo.h"

#line 2 "foo.cpp"

```

``` console
C:\Users\kimiyuki\Desktop\New folder>g++ foo.bundled.cpp
foo.bundled.cpp:1:9: error: incomplete universal character name \U
 #line 1 "C:\Users\kimiyuki\Desktop\foo.h"
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
foo.bundled.cpp:1:9: warning: unknown escape sequence: '\k'
foo.bundled.cpp:1:9: warning: unknown escape sequence: '\D'
```

--- a/onlinejudge_verify/languages/cplusplus_bundle.py
+++ b/onlinejudge_verify/languages/cplusplus_bundle.py
@@ -176,6 +176,7 @@ class Bundler(object):
         except ValueError:
             pass
         # パス中の特殊文字を JSON style にエスケープしてから生成コードに記述
+        # quick solution to this: https://github.com/online-judge-tools/verification-helper/issues/280
         self.result_lines.append('#line {} {}\n'.format(line, json.dumps(str(path))).encode())
 
     # path を解決する

--- a/onlinejudge_verify/languages/cplusplus_bundle.py
+++ b/onlinejudge_verify/languages/cplusplus_bundle.py
@@ -174,7 +174,7 @@ class Bundler(object):
             path = path.relative_to(pathlib.Path.cwd())
         except ValueError:
             pass
-        self.result_lines.append('#line {} "{}"\n'.format(line, str(path).replace("\\", "/")).encode())
+        self.result_lines.append('#line {} {}\n'.format(line, json.dumps(str(path))).encode())
 
     # path を解決する
     # see: https://gcc.gnu.org/onlinedocs/gcc/Directory-Options.html#Directory-Options

--- a/onlinejudge_verify/languages/cplusplus_bundle.py
+++ b/onlinejudge_verify/languages/cplusplus_bundle.py
@@ -1,5 +1,6 @@
 # Python Version: 3.x
 import functools
+import json
 import os
 import pathlib
 import re

--- a/onlinejudge_verify/languages/cplusplus_bundle.py
+++ b/onlinejudge_verify/languages/cplusplus_bundle.py
@@ -174,7 +174,7 @@ class Bundler(object):
             path = path.relative_to(pathlib.Path.cwd())
         except ValueError:
             pass
-        self.result_lines.append('#line {} "{}"\n'.format(line, str(path)).encode())
+        self.result_lines.append('#line {} "{}"\n'.format(line, str(path).replace("\\", "/")).encode())
 
     # path を解決する
     # see: https://gcc.gnu.org/onlinedocs/gcc/Directory-Options.html#Directory-Options

--- a/onlinejudge_verify/languages/cplusplus_bundle.py
+++ b/onlinejudge_verify/languages/cplusplus_bundle.py
@@ -175,6 +175,7 @@ class Bundler(object):
             path = path.relative_to(pathlib.Path.cwd())
         except ValueError:
             pass
+        # パス中の特殊文字を JSON style にエスケープしてから生成コードに記述
         self.result_lines.append('#line {} {}\n'.format(line, json.dumps(str(path))).encode())
 
     # path を解決する

